File: helloworld_web_war/src/main/java/org/example/demo/config/JsonConfigure.java
package org.example.demo.config;

import org.noear.snack.core.Feature;
import org.noear.solon.annotation.Bean;
import org.noear.solon.annotation.Configuration;
import org.noear.solon.serialization.snack3.SnackStringSerializer;

import java.math.BigDecimal;

@Configuration
public class JsonConfigure {
    @Bean
    public void init(SnackStringSerializer serializer){
        serializer.getSerializeConfig().addFeatures(Feature.EnumUsingName);

        serializer.addEncoder(Long.class, val -> String.valueOf(val));
        serializer.addEncoder(BigDecimal.class, val -> val.toPlainString());
    }
}

File: helloworld_web_jar_kt/src/main/kotlin/com/example/demo/config/JsonConfigure.kt
package com.example.demo.config

import org.noear.snack.core.Feature
import org.noear.solon.annotation.Bean
import org.noear.solon.annotation.Configuration
import org.noear.solon.annotation.Inject
import org.noear.solon.serialization.snack3.SnackRenderFactory
import java.math.BigDecimal


@Configuration
class JsonConfigure {
    @Bean
    fun init(serializer: SnackStringSerializer) {
        serializer.getSerializeConfig().addFeatures(Feature.EnumUsingName)

        serializer.addEncoder(Long::class.java) { `val`: Long -> `val`.toString() }
        serializer.addEncoder(BigDecimal::class.java) { `val`: BigDecimal -> `val`.toPlainString() }
    }
}